controller/src/main/java/org/jboss/as/controller/extension/ExtensionRegistry.java

```diff
@@ -258,10 +302,18 @@ public ExtensionContext getExtensionContext(final String moduleName, ManagementR
      * @throws IllegalStateException if the extension still has subsystems present in {@code rootResource} or its children
      */
     public void removeExtension(Resource rootResource, String moduleName, ManagementResourceRegistration rootRegistration) throws IllegalStateException {
-        // Can't use processType.isServer() to determine where to look for profile reg because a lot of test infrastructure
-        // doesn't add the profile mrr even in HC-based tests
-        ManagementResourceRegistration profileReg = rootRegistration.getSubModel(PathAddress.pathAddress(PathElement.pathElement(PROFILE)));
-        if (profileReg == null) {
+        final ManagementResourceRegistration profileReg;
+        if (rootRegistration.getPathAddress().size() == 0) {
```

Comment:
that is bit odd condition, why not something like getPathAddress() == PathAddress.EMPTY_ADDRESS